packages/loader/driver-utils/src/error.ts

```diff
@@ -0,0 +1,17 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IFluidErrorBase, LoggingError } from "@fluidframework/telemetry-utils";
+
+/** Error indicating an API is being used improperly resulting in an invalid operation. */
+export class UsageError extends LoggingError implements IFluidErrorBase {
+    readonly errorType = "usageError";
```

Comment:
We should not use strings, instead use values from one of the enums. Given that errors thrown is part of API contract (see such things are Container.close(), "closed" event), we need to ensure that clients can enumerate all known errorTypes different layers of our stack can throw. Anything unknown (i.e. not having shape of error our framework throws) is wrapped into generic error somewhere, so we maintain contract that client can enumerate all error types at complication time (assuming no dynamic code loading, of course)

Reply:
If I change the `DriverErrorType` enum won't it become a breaking change? It is also inside `driver-definitions`, which will make this a multi-step change. Or are you saying a create a new enum within `driver-utils`?

Reply:
No, I think adding it to DriverErrorType is the right move. I think you can do so as a next step, as long as it matches string in ContainerErrorType (as consumers do not know what enum was used for most part - all errors are consumed by consumers from loader layer).

Reply:
[AB#962](https://dev.azure.com/fluidframework/internal/_workitems/edit/962)